### PR TITLE
Support workflow restrictions in actions_runner_groups

### PR DESCRIPTION
**Pull request by bob-bins**
Fixes: #2558.

### CODE REVIEW

**Review by gmlewis — CHANGES_REQUESTED** [review]
Thank you, @bob-bins .
Please run "go generate ./..." and push the changes to the PR (we squash-and-merge all PRs) as described in CONTRIBUTING.md, and make sure to sign the CLA before we proceed.

**Comment by bob-bins** [comment]
@gmlewis I know the Contributors readme discourages force pushing, but I squashed and force-pushed when I corrected my commit signature. Thank you!

**Review by gmlewis — CHANGES_REQUESTED** [review]
Thank you, @bob-bins !
Please make sure to run `gofmt` on the this file and its test file, and push the results after making the tweaks below.

File: github/actions_runner_groups.go [hunk 1]
```diff
@@ -12,14 +12,16 @@ import (
...
+	ID                       *int64   `json:"id,omitempty"`
+	Name                     *string  `json:"name,omitempty"`
+	Visibility               *string  `json:"visibility,omitempty"`
+	Default                  *bool    `json:"default,omitempty"`
+	SelectedRepositoriesURL  *string  `json:"selected_repositories_url,omitempty"`
+	RunnersURL               *string  `json:"runners_url,omitempty"`
+	Inherited                *bool    `json:"inherited,omitempty"`
+	AllowsPublicRepositories *bool    `json:"allows_public_repositories,omitempty"`
+	RestrictedToWorkflows    *bool    `json:"restricted_to_workflows,omitempty"`
+	SelectedWorkflows        []string `json:"selected_workflows,omitempty"`
```
Comment (gmlewis) [inline github/actions_runner_groups.go:24]: While you are here, can you please also add the following field that is currently missing according to the official docs?

```go
WorkflowRestrictionsReadOnly *bool `json:"workflow_restrictions_read_only,omitempty"`
```

Comment (bob-bins) [reply github/actions_runner_groups.go]: I added this, but I can't actually find documentation on exactly what this param means so I'm unsure which tests cases I should add for it

**Pull request merged**